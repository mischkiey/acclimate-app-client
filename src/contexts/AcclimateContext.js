import React from 'react';

const AcclimateContext = React.createContext({});

export default AcclimateContext;