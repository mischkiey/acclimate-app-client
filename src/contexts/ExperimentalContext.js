import React, { Component } from 'react';
import TokenService from '../services/token-services';

export const ExperimentalContext = React.createContext({});

export class ExperimentalProvider extends Component {
    state = {
        log: false,
        tasks: [],
        shoppingItems: [],
    };

    handleUserLog = () => {
        const log = TokenService.hasAuthToken();
        this.setState({log});
    };

    handleUserLogOut = () => {
        TokenService.clearAuthToken();
        this.handleUserLog();
    };

    setTasks = (tasks) => {
        this.setState({tasks})
    };

    setShoppingItems = (shoppingItems) => {
        this.setState({shoppingItems})
    };

    render() {
        const value = {
            log: this.state.log,
            tasks: this.state.tasks,
            shoppingItems: this.state.shoppingItems,

            handleUserLog: this.handleUserLog,
            handleUserLogOut: this.handleUserLogOut,
            setTasks: this.setTasks,
            setShoppingItems: this.setShoppingItems,
        };

        return (
            <ExperimentalContext.Provider value={value}>
                {this.props.children}
            </ExperimentalContext.Provider>
        );
    };
};

