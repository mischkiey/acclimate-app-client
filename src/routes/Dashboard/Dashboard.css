.group {
    /* border: 1px dashed blue; */
}

.item {
    /* border: 1px dashed coral; */
}

.dashboard-component-group {
    /* border: 1px dashed violet; */
    /* border-radius: 10px; */
    display: flex;
    flex-flow: column;
    padding: 6px 6px 3px 6px;
    /* padding: 10px 10px 0px 10px; */
}

.disaster-utility-group {
    border: 1px dashed #37474f;
    border-radius: 10px;
    display: flex;
    flex-flow: column;
    padding: 5px;
    margin: 3px;
    /* padding: 10px; */
    /* margin-bottom: 10px; */
}

@media all and (min-width: 800px) {
    .dashboard-component-group {
        /* border: 1px dashed violet; */
        display: flex;
        flex-flow: column;
        padding: 10px 10px 5px 10px;
        /* padding: 10px 10px 0px 10px; */
    }
    
    .disaster-utility-group {
        border: 1px dashed #37474f;
        display: flex;
        flex-flow: column;
        padding: 5px;
        margin: 5px;
        /* padding: 10px; */
        /* margin-bottom: 10px; */
    }
}