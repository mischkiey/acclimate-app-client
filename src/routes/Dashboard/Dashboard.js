import React, { Component } from 'react';
import APIService from '../../services/api-services';
import TokenService from '../../services/token-services';
import DisasterProgram from '../../components/DisasterProgram/DisasterProgram';
import TaskList from '../../components/TaskList/TaskList';
import ShoppingList from '../../components/ShoppingList/ShoppingList';
import './Dashboard.css';

class Dashboard extends Component {
    state = {
        programs: [],
        error: null,
    };

    componentDidMount() {
        this.setState({error: null})

        const token = TokenService.getAuthToken();

        return APIService.get('/disaster/user/program', token)
            .then(programs => {
                this.setState({programs})
            })
            .catch(({error}) =>{
                console.log(error, 'Dashboard Error')
                this.setState({error})
            })
    };

    render () {
        const programs = this.state.programs.map(program => <DisasterProgram key={program.disaster_program_id} {...program} />)

        return (
            <div className='group'> 
                <section className='dashboard-component-group item'>
                    {programs}
                </section>

                <section className='dashboard-component-group item'>
                    <article className='disaster-utility-group item'>
                        <TaskList />
                    </article>

                    <article className='disaster-utility-group item'>
                        <ShoppingList />
                    </article>
                </section>
            </div>
        );
    };
};

export default Dashboard;