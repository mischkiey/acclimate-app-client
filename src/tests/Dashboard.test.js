import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { AcclimateProvider } from '../contexts/AcclimateContext';
import Dashboard from '../routes/Dashboard/Dashboard';

describe(`Dashboard Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <AcclimateProvider>
                        <Dashboard />
                    </AcclimateProvider>
                </BrowserRouter>,
                div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <AcclimateProvider>
                            <Dashboard />
                        </AcclimateProvider>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})