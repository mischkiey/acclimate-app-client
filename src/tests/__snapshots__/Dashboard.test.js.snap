// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="group"
>
  <section
    className="dashboard-components-group item"
  />
  <section
    className="dashboard-components-group item"
  >
    <article
      className="dashboard-component-group item"
    >
      <h2
        className="center"
      >
        Task List
      </h2>
      <form
        className="list-item-group"
        onSubmit={[Function]}
      >
        <input
          className="list-item-input"
          name="add_user_task"
          placeholder="Enter task"
          type="text"
        />
        <button
          className="tooltip y-btn"
        >
          <i
            className="material-icons"
          >
            add_circle
          </i>
          <span
            className="long tooltiptext"
          >
            Click to add to your task list
          </span>
        </button>
      </form>
    </article>
    <article
      className="dashboard-component-group item"
    >
      <h2
        className="center"
      >
        Shopping List
      </h2>
      <form
        className="list-item-group"
        onSubmit={[Function]}
      >
        <input
          className="list-item-input"
          name="add_user_shopping_item"
          placeholder="Enter shopping item"
          type="text"
        />
        <button
          className="tooltip y-btn"
        >
          <i
            className="material-icons"
          >
            add_circle
          </i>
          <span
            className="long tooltiptext"
          >
            Click to add item to your shopping list
          </span>
        </button>
      </form>
    </article>
  </section>
</div>
`;
